File: 0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        result = 0
        temp = x
        if temp < 0:
            x *= -1
        while x != 0:
            result += x % 10
            x //= 10
            if x > 0:
                result *= 10
                
        if temp < 0:
            result *= -1
            
        if result >= 2**31 or result < -1*(2**31):
            return 0
        return result
            

File: 0007-reverse-integer/solution.cpp
class Solution {
public:
    int reverse(int x) 
    {
        if(x == INT_MIN)
            return 0;
        bool isNeg = false;
        if((x>>31)&1){
            isNeg = true;
            x = x*-1;   
        }
        
        
        
        int result = 0;
        
        while(x)
        {
            result += x%10;
            
            // Check overflow
            if(result > INT_MAX/10 && x >= 10)
            {
                return 0;
            }
            if(x >= 10)
            {
                result *= 10;
            }
            x /= 10;
        }
        
        if(isNeg)
            result *= -1;
        return result;
        
        
    }
};
